### ISSUE TITLE
[NODE] Dockerfile for arm

### ISSUE THREAD

**Issue by filipmacek** [smartcontractkit/chainlink#3227]
Hello, is there docker image for chainlink node that can run on arm cpu ( like Raspberry Pi's)
Thanks,
Love this project,writing my master thesis about chainlink

**Comment by willianpaixao** [comment]
Related #3596 

Hi @filipmacek, I would like to share my recent experience running a ChainLink full node on Raspberry Pi 4:
1. First I tried to cross-compile the binary from my laptop, it almost worked but the Makefile got confused since it also compiles Ethereum's Go library and it didn't work. I think it's possible but the `GNUMakefile` would need good refactoring to account for cross-compilation.
2. Then I installed Go, Yarn and NodeJS in the RPI, which I didn't want to compile ChainLink natively in the ARM64 architecture. It works, but the compilation logs show a bigger amount of warnings and even errors than when I compiled in my AMD64. I was able to run the binary, however didn't check the extent of the errors/warnings.

I also created issue #3592 so we can simply download the pre-compiled (even audited) binary, as I do with Go, Geth, and many other software.
Although my goal to run ChainLink in the RPi without Docker to squeeze a drop of performance, you're right, ChainLink should also generate images in other architectures. This can be archived in a few ways, like using a build pipeline/CICD tool (eg Travis, etc) or through [Docker buildx](https://github.com/docker/buildx/) plugin.
